Eshop.WebApi/Controllers/ProductController.cs

```diff
@@ -7,50 +8,56 @@ namespace Eshop.WebApi.Controllers
     [Route("[controller]")]
     public class ProductController : Controller
     {
-        private static List<Product> Products = new List<Product>
+        private readonly EshopDbContext dbContext;
+
+        public ProductController(EshopDbContext dbContext)
         {
-            new Product(1, "Notebook Acer 16", "Best notebook out there", 399.99m, CategoryController.Categories[0]),
-            new Product(2, "Mouse Razor 123", "Best Mouse", 14.50m, CategoryController.Categories[1]),
-        };
+            this.dbContext = dbContext;
+        }
 
         [HttpGet]
         public List<Product> GetProducts()
         {
-            return Products;
+            return dbContext.Products.ToList();
         }
 
         [HttpGet("{id}")]
         public Product GetProduct(int id)
         {
-            return Products.First(x => x.Id == id);
+            return dbContext.Products.First(x => x.Id == id);
         }
 
         [HttpPost]
         public Product CreateProduct(string title, string description, decimal price, int categoryId)
         {
             var category = CategoryController.Categories.First(x => x.Id == categoryId);
-            var newId = Products.Max(x => x.Id) + 1;
+            var newId = dbContext.Products.Max(x => x.Id) + 1;
```

Comment:
this is not the same as Category
actually you ruined Category - check comment about newId there